<!-- 
RSS generated by JIRA (6.3.4#6332-sha1:51bc225ef474afe3128b2f66878477f322397b16) at Fri Dec 16 19:15:17 UTC 2016

It is possible to restrict the fields that are returned in this document by specifying the 'field' parameter in your request.
For example, to request only the issue key and summary add field=key&field=summary to the URL of your request.
For example:
https://issues.apache.org/jira/si/jira.issueviews:issue-xml/HBASE-4030/HBASE-4030.xml?field=key&amp;field=summary
-->
<rss version="0.92" >
<channel>
    <title>ASF JIRA</title>
    <link>https://issues.apache.org/jira</link>
    <description>This file is an XML representation of an issue</description>
    <language>en-uk</language>    <build-info>
        <version>6.3.4</version>
        <build-number>6332</build-number>
        <build-date>15-08-2014</build-date>
    </build-info>

<item>
            <title>[HBASE-4030] LoadIncrementalHFiles fails with FileNotFoundException</title>
                <link>https://issues.apache.org/jira/browse/HBASE-4030</link>
                <project id="12310753" key="HBASE">HBase</project>
                    <description>&lt;p&gt;&amp;#8211; We&apos;ve been seeing intermittent failures of calls to LoadIncrementalHFiles.  When this happens the node that made the call will see a FileNotFoundException such as this:&lt;/p&gt;

&lt;p&gt;2011-06-23 15:47:34.379566500 java.net.SocketTimeoutException: Call to s8.XXX/67.215.90.38:60020 failed on socket timeout exception: java.net.SocketTi&lt;br/&gt;
meoutException: 60000 millis timeout while waiting for channel to be ready for read. ch : java.nio.channels.SocketChannel[connected local=/67.215.90.51:51605 remo&lt;br/&gt;
te=s8.XXX/67.215.90.38:60020]&lt;br/&gt;
2011-06-23 15:47:34.379570500 java.io.FileNotFoundException: java.io.FileNotFoundException: File does not exist: /hfiles/2011/06/23/14/domainsranked/TopDomainsRan&lt;br/&gt;
k.r3v5PRvK/handling/3557032074765091256&lt;br/&gt;
2011-06-23 15:47:34.379573500   at org.apache.hadoop.hdfs.DFSClient$DFSInputStream.openInfo(DFSClient.java:1602)&lt;br/&gt;
2011-06-23 15:47:34.379573500   at org.apache.hadoop.hdfs.DFSClient$DFSInputStream.&amp;lt;init&amp;gt;(DFSClient.java:1593)&lt;/p&gt;

&lt;p&gt;&amp;#8211; Over on the regionserver that was loading this we see that it attempted to load and hit a 60 second timeout:&lt;/p&gt;

&lt;p&gt;2011-06-23 15:45:54,634 INFO org.apache.hadoop.hbase.regionserver.Store: Validating hfile at hdfs://namenode.XXX/hfiles/2011/06/23/14/domainsranked/TopDomainsRank.r3v5PRvK/handling/3557032074765091256 for inclusion in store handling region domainsranked,368449:2011/0/03/23:category:ffffffff:com.zynga.static.fishville.facebook,1305890318961.d4925aca7852bed32613a509215d42b&lt;br/&gt;
8.&lt;br/&gt;
...&lt;br/&gt;
2011-06-23 15:46:54,639 INFO org.apache.hadoop.hdfs.DFSClient: Failed to connect to /67.215.90.38:50010, add to deadNodes and continue&lt;br/&gt;
java.net.SocketTimeoutException: 60000 millis timeout while waiting for channel to be ready for read. ch : java.nio.channels.SocketChannel&lt;span class=&quot;error&quot;&gt;&amp;#91;connected local=/67.215.90.38:42199 remote=/67.215.90.38:50010&amp;#93;&lt;/span&gt;&lt;br/&gt;
at org.apache.hadoop.net.SocketIOWithTimeout.doIO(SocketIOWithTimeout.java:164)&lt;br/&gt;
at org.apache.hadoop.net.SocketInputStream.read(SocketInputStream.java:155)&lt;br/&gt;
at org.apache.hadoop.net.SocketInputStream.read(SocketInputStream.java:128)&lt;br/&gt;
at java.io.BufferedInputStream.fill(BufferedInputStream.java:218)&lt;br/&gt;
at java.io.BufferedInputStream.read(BufferedInputStream.java:237)&lt;br/&gt;
at java.io.DataInputStream.readShort(DataInputStream.java:295)&lt;/p&gt;

&lt;p&gt;&amp;#8211; We suspect this particular problem is a resource contention issue on our side.  However, the loading process proceeds to rename the file despite the failure:&lt;/p&gt;

&lt;p&gt;2011-06-23 15:46:54,657 INFO org.apache.hadoop.hbase.regionserver.Store: Renaming bulk load file hdfs://namenode.XXX/hfiles/2011/06/23/14/domainsranked/TopDomainsRank.r3v5PRvK/handling/3557032074765091256 to hdfs://namenode.XXX:8020/hbase/domainsranked/d4925aca7852bed32613a509215d42b8/handling/3615917062821145533&lt;/p&gt;

&lt;p&gt;&amp;#8211; And then the LoadIncrementalHFiles tries to load the hfile again:&lt;/p&gt;

&lt;p&gt;2011-06-23 15:46:55,684 INFO org.apache.hadoop.hbase.regionserver.Store: Validating hfile at hdfs://namenode.XXX/hfiles/2011/06/23/14/domainsranked/TopDomainsRank.r3v5PRvK/handling/3557032074765091256 for inclusion in store handling region domainsranked,368449:2011/05/03/23:category:ffffffff:com.zynga.static.fishville.facebook,1305890318961.d4925aca7852bed32613a509215d42b8.&lt;/p&gt;

&lt;p&gt;2011-06-23 15:46:55,685 DEBUG org.apache.hadoop.ipc.HBaseServer: IPC Server handler 147 on 60020, call bulkLoadHFile(hdfs://namenode.XXX/hfiles/2011/06/23/14/domainsranked/TopDomainsRank.r3v5PRvK/handling/3557032074765091256, [B@4224508b, [B@5e23f799) from 67.215.90.51:51856: error: java.io.FileNotFoundException: File does not exist: /hfiles/2011/06/23/14/domainsranked/TopDomainsRank.r3v5PRvK/handling/3557032074765091256&lt;/p&gt;

&lt;p&gt;&amp;#8211; This eventually leads to the load command failing.&lt;/p&gt;</description>
                <environment>&lt;p&gt;CDH3bu on Ubuntu 4.4.3&lt;/p&gt;</environment>
        <key id="12511494">HBASE-4030</key>
            <summary>LoadIncrementalHFiles fails with FileNotFoundException</summary>
                <type id="1" iconUrl="https://issues.apache.org/jira/images/icons/issuetypes/bug.png">Bug</type>
                                            <priority id="3" iconUrl="https://issues.apache.org/jira/images/icons/priorities/major.png">Major</priority>
                        <status id="5" iconUrl="https://issues.apache.org/jira/images/icons/statuses/resolved.png" description="A resolution has been taken, and it is awaiting verification by reporter. From here issues are either reopened, or are closed.">Resolved</status>
                    <statusCategory id="3" key="done" colorName="green"/>
                                    <resolution id="2">Won&apos;t Fix</resolution>
                                        <assignee username="-1">Unassigned</assignee>
                                    <reporter username="amp">Adam Phelps</reporter>
                        <labels>
                    </labels>
                <created>Fri, 24 Jun 2011 16:58:28 +0000</created>
                <updated>Sun, 6 Jul 2014 15:11:52 +0000</updated>
                            <resolved>Tue, 28 Aug 2012 22:51:21 +0000</resolved>
                                    <version>0.90.1</version>
                                                        <due></due>
                            <votes>0</votes>
                                    <watches>6</watches>
                                                                <comments>
                            <comment id="13443613" author="lhofhansl" created="Tue, 28 Aug 2012 22:51:21 +0000"  >&lt;p&gt;This has been open for a while without any activity. Closing.&lt;br/&gt;
Please reopen if you disagree.&lt;/p&gt;</comment>
                            <comment id="13763305" author="pjsg" created="Tue, 10 Sep 2013 18:06:06 +0000"  >&lt;p&gt;This has started to bite us in 0.94.8 &amp;#8211; fairly consistently. I have no idea what changed in our environment to trigger this behavior.&lt;/p&gt;</comment>
                            <comment id="14053136" author="amitsela" created="Sun, 6 Jul 2014 15:11:52 +0000"  >&lt;p&gt;Happened to me in 0.94.12 - I can&apos;t seem to reproduce.&lt;br/&gt;
It happened &quot;out of nowhere&quot; - no change made in application/infrastructure in a while.&lt;br/&gt;
Audit log shows that the same regionserver is opening one of the regions, renaming (moving from MR output dir into hbase region directory) and trying to open again from MR output dir (repeating 10 times).&lt;br/&gt;
Open-Rename-10xOpen  appears in that order in the audit log, with a msec difference all in the same region server.&lt;/p&gt;

&lt;p&gt;Logs:&lt;br/&gt;
On the client side this is the Excpetion:&lt;br/&gt;
java.net.SocketTimeoutException: Call to node.xxx.com/xxx.xxx.xxx.xxx:PORT failed on socket timeout exception: java.net.SocketTimeoutException: 60000 millis timeout while waiting for channel to be ready for read. ch : java.nio.channels.SocketChannel&lt;span class=&quot;error&quot;&gt;&amp;#91;connected local=/xxx.xxx.xxx.xxx:PORT remote=node.xxx.com/xxx.xxx.xxx.xxx:PORT&amp;#93;&lt;/span&gt;&lt;br/&gt;
org.apache.hadoop.hbase.mapreduce.LoadIncrementalHFiles$3@29f2a6e3, org.apache.hadoop.ipc.RemoteException: org.apache.hadoop.io.MultipleIOException: 6 exceptions &lt;span class=&quot;error&quot;&gt;&amp;#91;java.io.FileNotFoundException: File does not exist: /data/output_jobs/output_websites/HFiles_20140705/metadata/88fd743853cf4f8a862fb19646027a48, java.io.FileNotFoundException: File does not exist: /data/output_jobs/output_websites/HFiles_20140705/gen/31c4c5cea9b348dbb6bb94115a483877, java.io.FileNotFoundException: File does not exist: /data/output_jobs/output_websites/HFiles_20140705/gen/5762c45aaf4f408ba748a989f7be9647, java.io.FileNotFoundException: File does not exist: /data/output_jobs/output_websites/HFiles_20140705/gen1/2ee02a005b654704a092d16c5c713373, java.io.FileNotFoundException: File does not exist: /data/output_jobs/output_websites/HFiles_20140705/gen1/618251330a1842a797de4b304d341a02, java.io.FileNotFoundException: File does not exist: /data/output_jobs/output_websites/HFiles_20140705/metadata/3955039392ce4f49aee5f58218a61be1&amp;#93;&lt;/span&gt;&lt;br/&gt;
	at org.apache.hadoop.io.MultipleIOException.createIOException(MultipleIOException.java:47)&lt;br/&gt;
	at org.apache.hadoop.hbase.regionserver.HRegion.bulkLoadHFiles(HRegion.java:3673)&lt;br/&gt;
	at org.apache.hadoop.hbase.regionserver.HRegion.bulkLoadHFiles(HRegion.java:3622)&lt;br/&gt;
	at org.apache.hadoop.hbase.regionserver.HRegionServer.bulkLoadHFiles(HRegionServer.java:2930)&lt;br/&gt;
	at sun.reflect.GeneratedMethodAccessor70.invoke(Unknown Source)&lt;br/&gt;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)&lt;br/&gt;
	at java.lang.reflect.Method.invoke(Method.java:601)&lt;br/&gt;
	at org.apache.hadoop.hbase.ipc.WritableRpcEngine$Server.call(WritableRpcEngine.java:320)&lt;br/&gt;
	at org.apache.hadoop.hbase.ipc.HBaseServer$Handler.run(HBaseServer.java:1426)&lt;/p&gt;

&lt;p&gt;On the regionserver:&lt;/p&gt;

&lt;p&gt;ERROR org.apache.hadoop.hbase.regionserver.HRegion: There were one or more IO errors when checking if the bulk load is ok.&lt;br/&gt;
org.apache.hadoop.io.MultipleIOException: 6 exceptions &lt;span class=&quot;error&quot;&gt;&amp;#91;java.io.FileNotFoundException: File does not exist: /data/output_jobs/output_websites/HFiles_20140705/metadata/88fd743853cf4f8a862fb19646027a48, java.io.FileNotFoundException: File does not exist: /data/output_jobs/output_websites/HFiles_20140705/gen/31c4c5cea9b348dbb6bb94115a483877, java.io.FileNotFoundException: File does not exist: /data/output_jobs/output_websites/HFiles_20140705/gen/5762c45aaf4f408ba748a989f7be9647, java.io.FileNotFoundException: File does not exist: /data/output_jobs/output_websites/HFiles_20140705/gen1/2ee02a005b654704a092d16c5c713373, java.io.FileNotFoundException: File does not exist: /data/output_jobs/output_websites/HFiles_20140705/gen1/618251330a1842a797de4b304d341a02, java.io.FileNotFoundException: File does not exist: /data/output_jobs/output_websites/HFiles_20140705/metadata/3955039392ce4f49aee5f58218a61be1&amp;#93;&lt;/span&gt;&lt;br/&gt;
        at org.apache.hadoop.io.MultipleIOException.createIOException(MultipleIOException.java:47)&lt;br/&gt;
        at org.apache.hadoop.hbase.regionserver.HRegion.bulkLoadHFiles(HRegion.java:3673)&lt;br/&gt;
        at org.apache.hadoop.hbase.regionserver.HRegion.bulkLoadHFiles(HRegion.java:3622)&lt;br/&gt;
        at org.apache.hadoop.hbase.regionserver.HRegionServer.bulkLoadHFiles(HRegionServer.java:2930)&lt;br/&gt;
        at sun.reflect.GeneratedMethodAccessor70.invoke(Unknown Source)&lt;br/&gt;
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)&lt;br/&gt;
        at java.lang.reflect.Method.invoke(Method.java:601)&lt;br/&gt;
        at org.apache.hadoop.hbase.ipc.WritableRpcEngine$Server.call(WritableRpcEngine.java:320)&lt;br/&gt;
        at org.apache.hadoop.hbase.ipc.HBaseServer$Handler.run(HBaseServer.java:1426)&lt;/p&gt;

&lt;p&gt;followed by:&lt;/p&gt;

&lt;p&gt;ERROR org.apache.hadoop.hbase.regionserver.HRegionServer: &lt;br/&gt;
org.apache.hadoop.hbase.ipc.CallerDisconnectedException: Aborting call next(4522610431482097770, 250), rpc version=1, client version=29, methodsFingerPrint=-1368823753 from xxx.xxx.xxx.xxx after 12507 ms, since caller disconnected&lt;br/&gt;
        at org.apache.hadoop.hbase.ipc.HBaseServer$Call.throwExceptionIfCallerDisconnected(HBaseServer.java:436)&lt;br/&gt;
        at org.apache.hadoop.hbase.regionserver.HRegion$RegionScannerImpl.nextInternal(HRegion.java:3980)&lt;br/&gt;
        at org.apache.hadoop.hbase.regionserver.HRegion$RegionScannerImpl.nextRaw(HRegion.java:3890)&lt;br/&gt;
        at org.apache.hadoop.hbase.regionserver.HRegion$RegionScannerImpl.nextRaw(HRegion.java:3880)&lt;br/&gt;
        at org.apache.hadoop.hbase.regionserver.HRegionServer.next(HRegionServer.java:2648)&lt;br/&gt;
        at sun.reflect.GeneratedMethodAccessor60.invoke(Unknown Source)&lt;br/&gt;
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)&lt;br/&gt;
        at java.lang.reflect.Method.invoke(Method.java:601)&lt;br/&gt;
        at org.apache.hadoop.hbase.ipc.WritableRpcEngine$Server.call(WritableRpcEngine.java:320)&lt;br/&gt;
        at org.apache.hadoop.hbase.ipc.HBaseServer$Handler.run(HBaseServer.java:1426)&lt;br/&gt;
2014-07-06 03:52:14,278 &lt;span class=&quot;error&quot;&gt;&amp;#91;IPC Server handler 28 on 8041&amp;#93;&lt;/span&gt; ERROR org.apache.hadoop.hbase.regionserver.HRegionServer: &lt;br/&gt;
org.apache.hadoop.hbase.ipc.CallerDisconnectedException: Aborting call next(7354511084312054096, 250), rpc version=1, client version=29, methodsFingerPrint=-1368823753 from xxx.xxx.xxx.xxx after 9476 ms, since caller disconnected&lt;br/&gt;
        at org.apache.hadoop.hbase.ipc.HBaseServer$Call.throwExceptionIfCallerDisconnected(HBaseServer.java:436)&lt;br/&gt;
        at org.apache.hadoop.hbase.regionserver.HRegion$RegionScannerImpl.nextInternal(HRegion.java:3980)&lt;br/&gt;
        at org.apache.hadoop.hbase.regionserver.HRegion$RegionScannerImpl.nextRaw(HRegion.java:3890)&lt;br/&gt;
        at org.apache.hadoop.hbase.regionserver.HRegion$RegionScannerImpl.nextRaw(HRegion.java:3880)&lt;br/&gt;
        at org.apache.hadoop.hbase.regionserver.HRegionServer.next(HRegionServer.java:2648)&lt;br/&gt;
        at sun.reflect.GeneratedMethodAccessor60.invoke(Unknown Source)&lt;br/&gt;
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)&lt;br/&gt;
        at java.lang.reflect.Method.invoke(Method.java:601)&lt;br/&gt;
        at org.apache.hadoop.hbase.ipc.WritableRpcEngine$Server.call(WritableRpcEngine.java:320)&lt;br/&gt;
        at org.apache.hadoop.hbase.ipc.HBaseServer$Handler.run(HBaseServer.java:1426) &lt;/p&gt;</comment>
                    </comments>
                    <attachments>
                    </attachments>
                <subtasks>
                    </subtasks>
                <customfields>
                                                <customfield id="customfield_12310310" key="com.atlassian.jira.toolkit:attachments">
                        <customfieldname>Attachment count</customfieldname>
                        <customfieldvalues>
                            <customfieldvalue>0.0</customfieldvalue>
                        </customfieldvalues>
                    </customfield>
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                <customfield id="customfield_12310220" key="com.atlassian.jira.ext.charting:firstresponsedate">
                        <customfieldname>Date of First Response</customfieldname>
                        <customfieldvalues>
                            <customfieldvalue>Tue, 28 Aug 2012 22:51:21 +0000</customfieldvalue>

                        </customfieldvalues>
                    </customfield>
                                                                                                                                                                                                                                                                                                                                                                                                                                            <customfield id="customfield_12310420" key="com.pyxis.greenhopper.jira:gh-global-rank">
                        <customfieldname>Global Rank</customfieldname>
                        <customfieldvalues>
                            <customfieldvalue>27137</customfieldvalue>
                        </customfieldvalues>
                    </customfield>
                                                                                                                                                    <customfield id="customfield_12312521" key="com.atlassian.jira.toolkit:LastCommentDate">
                        <customfieldname>Last public comment date</customfieldname>
                        <customfieldvalues>
                            2 years, 23 weeks, 5 days ago
                        </customfieldvalues>
                    </customfield>
                                                                                            <customfield id="customfield_12311820" key="com.pyxis.greenhopper.jira:gh-lexo-rank">
                        <customfieldname>Rank</customfieldname>
                        <customfieldvalues>
                            <customfieldvalue>0|i0hp2n:</customfieldvalue>

                        </customfieldvalues>
                    </customfield>
                                                                <customfield id="customfield_12310920" key="com.pyxis.greenhopper.jira:gh-global-rank">
                        <customfieldname>Rank (Obsolete)</customfieldname>
                        <customfieldvalues>
                            <customfieldvalue>101324</customfieldvalue>
                        </customfieldvalues>
                    </customfield>
                                                                                                                                                                                                                                                                                                <customfield id="customfield_12310222" key="com.atlassian.jira.ext.charting:timeinstatus">
                        <customfieldname>Time in Status</customfieldname>
                        <customfieldvalues>
                            
                        </customfieldvalues>
                    </customfield>
                                                                                                                        </customfields>
    </item>
</channel>
</rss>